src/Controller/AdController.php

```diff
@@ -18,11 +25,252 @@ public function index()
+            $ad = $this->getDoctrine()->getRepository(Ad::class)->find($id);
+            // Set all variables for a new message
+            $message -> setSentTo($ad->getCreatedBy());
+            $message -> setSentFrom($this->getUser());
+            $message -> setWrittenAt(new \DateTime());
+            // Post the message to DB
+            $entityManager = $this->getDoctrine()->getManager();
+            $entityManager->persist($message);
+            $entityManager->flush();
+            // Send an email to the message receiver about the message
+            $emailMessage = (new \Swift_Message('Message to seller'))
+                ->setFrom('bemantelio@gmail.com')
+                ->setTo($ad->getCreatedBy()->getEmail())
+                ->setBody(
+                    $this->renderView(
+                        'message/email-message.html.twig',
+                        ['userEmail' => $ad->getCreatedBy()->getEmail()]
+                    ),
+                    'text/html'
+                );
+            //$mailer->send($emailMessage);
+            // Flash a success message
+            $this->addFlash('success', 'Your message was sent');
+            // Render view template
+            return $this->redirectToRoute('view_ad', array('id' => $id));
+        }
+        // Flash a warning message
+        $this->addFlash('warning', 'Something went wrong');
+        // Render view template
+        return $this->redirectToRoute('view_ad', array('id' => $id));
+    }
+
+    /**
+     * @Route(path="/ads/{id}/{commid}/edit", name="editcomment")
+     */
+    public function editcomment(Request $request, $id, $commid)
+    {
+        // Get {commid} comment
+        $comment = $this->getDoctrine()->getRepository(Comment::class)->find($commid);
+        // Form for existing comment to edit
+        $form = $this->createForm(CommentType::class, $comment);
+        // Render contact template
+        return $this->render('ad/edit-comment.html.twig', [
+            'id' => $id,    // Send ad {id} to edit-comment template
+            'commid' => $commid,    // Send comment {commid} to edit-comment template
+            'form' => $form->createView(),  // Send created form to reply template
+            'error' => $form->getErrors(true)
+        ]);
+    }
+
+    /**
+     * @Route(path="/ads/{id}/{commid}/edit/success", name="submitedit")
+     */
+    public function editcommentsubmit(Request $request, $id, $commid)
+    {
+        // Get {commid} comment
+        $comment = $this->getDoctrine()->getRepository(Comment::class)->find($commid);
+        // Form for existing comment to edit
+        $form = $this->createForm(CommentType::class, $comment);
+        $form->handleRequest($request);
+        // Check if the comment was written by current {user}
+        if ($comment->getWrittenBy()->getId() == $this->getUser()->getId()) {
+            // Check if form was correctly filled
+            if ($form->isSubmitted() && $form->isValid()) {
```

Comment:
vėlgi nereikia atskiro path

Reply:
Not 100% sure what do you mean by that. Čia tipo vėl to pačio tikrinimo? in that case, same as reply ir contact.